### PR TITLE
Add arrow navigation to Warning dropdown menu

### LINKED ISSUE #24303: Invalid block warning dropdown menu cannot be navigated using arrow keys.
**Describe the bug**
When a user has an invalid block, the extra options in the dropdown cannot be navigated using a keyboard. The dropdown closes and focus is moved in an unexpected way.

Possibly another issue with `WritingFlow` interrupting keyboard navigation.

**To reproduce**
1. Add a block and make it invalid by switching to the code editor and editing the markup (e.g. changing the tags)
2. Try opening the additional options using the keyboard and navigating between the options using arrow keys.
3. Observe the dropdown menu closes unexpectedly and focus jumps to elsewhere in the editor

**Expected behavior**
The user can navigate between the options and the menu stays open.

**Screenshots**
Screenshot 2020-07-31 at 1 56 36 pm

### PR DESCRIPTION
**Pull request by ntsekouras**
## Description

Fixes: https://github.com/WordPress/gutenberg/issues/24303

This PR refactors `Warning` component to use `DropdownMenu` instead of `Dropdown` to enable arrow navigation in its secondary menu.

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/block-editor/src/components/warning/index.js [hunk 1]
```diff
@@ -29,19 +34,16 @@ function Warning( { className, actions, children, secondaryActions } ) {
...
-								renderToggle={ ( { isOpen, onToggle } ) => (
-									<Button
-										icon={ moreHorizontal }
-										label={ __( 'More options' ) }
-										onClick={ onToggle }
-										aria-expanded={ isOpen }
-									/>
-								) }
-								renderContent={ () => (
+								icon={ <BlockIcon icon={ moreHorizontal } /> }
```
Comment (youknowriad) [inline packages/block-editor/src/components/warning/index.js:39]: Why `BlockIcon`, can we use the svg `moreHorizontal` directly as `icon`? If not, I think it's better to use `Icon from '@wordpress/icons` component.

Comment (ntsekouras) [reply packages/block-editor/src/components/warning/index.js]: We can. I had the impression the usage of `BlockIcon` was preferred from some other code I had seen in the past.. Will investigate more the differences of them.

Changed it. Thanks!

Comment (youknowriad) [reply packages/block-editor/src/components/warning/index.js]: I think BlockIcon is for "block" icons (icons used for block registration). Block icons support some color properties there...

Comment (ntsekouras) [reply packages/block-editor/src/components/warning/index.js]: Yes you're right! And internally uses the `Icon` component as you proposed. 💯

File: packages/block-editor/src/components/warning/style.scss [hunk 2]
```diff
@@ -50,4 +50,7 @@
 	// Set z-index as if it's displayed on the bottom, otherwise the modal
 	// dialog popover might overlap if displayed on the bottom.
 	z-index: z-index(".components-popover.block-editor-warning__dropdown");
+	.components-menu-item__button {
+		padding: 6px 12px;
```
Comment (youknowriad) [inline packages/block-editor/src/components/warning/style.scss:54]: I think you don't need this if you pass a `noIcons` prop to `DropdownMenu`

**Review by youknowriad — APPROVED** [review]
Left a small comment but  LGTM 👍

**Comment by talldan** [comment]
Thanks for fixing this, Nik!

**Pull request merged**